### ISSUE TITLE
Client-side compiling needs attention

### ISSUE THREAD

**Issue by niftylettuce** [pugjs/pug#2974]
Currently there's dozens of unmaintained and broken packages that deal with using Pug on the client.

For example, some packages don't support mixins, while others don't pass the `filename` option to the function to compile the template so that includes/blocks would work.  Others are using old versions of Pug known as Jade, some have broken peer dependencies, and lastly some simply don't work.

* https://github.com/gulp-community/gulp-pug
* https://github.com/izolate/pugify
* https://github.com/leegcc/pugify
* https://github.com/drfisher/gulp-pug-template-concat

**Comment by niftylettuce** [comment]
My core issue is https://github.com/pugjs/pug/issues/1773#issuecomment-370680147.

**Comment by ForbesLindesay** [comment]
Yes, I'd like to see us move some of these into the pug org (where appropriate), and add documentation about them to our website.

**Comment by cekvenich** [comment]
also pug-cli, the one I use.
